### PR TITLE
Make compatible with clang and ThreadSanitizer

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

**Comment by erikcorry** [comment]
```
export CXX=clang++
export CC=clang
export CFLAGS=-fsanitize=thread\ -D__SANITIZE_THREAD__=1
export CXXFLAGS=-fsanitize=thread\ -D__SANITIZE_THREAD__=1
export LDFLAGS=-fsanitize=thread
export BUILD_TYPE=Debug
rm -rf build
make
```

File: src/third_party/dartino/object_memory_mark_sweep.cc [hunk 1]
```diff
@@ -630,7 +630,6 @@ void OldSpace::validate_sweep(Chunk* chunk) {
     ASSERT(GcMetadata::all_mark_bits_are(object, size, alive ? 1 : 0));
     ASSERT(object->is_a_free_object() == !alive);
     if (*starts != GcMetadata::NO_OBJECT_START) {
-      ASSERT(*starts < GcMetadata::CARD_SIZE);
```
Comment (kasperl) [inline src/third_party/dartino/object_memory_mark_sweep.cc:633]: Not sure I understand why this had to go?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/third_party/dartino/object_memory_mark_sweep.cc (hunk 1 above)
Comment (erikcorry) [reply src/third_party/dartino/object_memory_mark_sweep.cc:633]: On 32 bit it was wrong, and on 64 bit it could never trigger.

**Pull request merged**